pages/k8s/ipv6.md

```diff
@@ -0,0 +1,182 @@
+        ports:
+        - containerPort: 80
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: nginx6
+  labels:
+    run: nginxdualstack
+spec:
+  type: NodePort
+  ipFamily: IPv6
+  ports:
+  - port: 80
+    protocol: TCP
+  selector:
+    run: nginxdualstack
+```
+
+<a id='known-issues> </a>
+## Known Issues
+
+Because of the pre-release feature status for IPv6 and dual-stack in Kubernetes
+and since Juju does not officially support IPv6, there are currently
+some things which won't work 100% and some limitations on IPv6 configuration. 
+These will also vary depending on the underlying cloud provider.
+
+### Juju
+
+The following arise because Juju does not fully support IPv6:
+
+* The charms require IPv4 on the underlying hosts, even when running a cluster
+  in IPv6-only mode.
+
+* By default, connections to the API server will use the IPv4 address even when
+  running a cluster in IPv6-preferred or IPv6-only mode. This can be modified
+  in the client config by hand or overridden via the `loadbalancer-ips` config
+  on the [kubernetes-master] and / or kubeapi-load-balancer charms.
+
+* IPv6 NodePort listeners won't function on the master, though they will work
+  on the worker units.
+
+* Juju's `open-port` cannot be used to allow NodePort connections for IPv6.
+
+### AWS
+
+The following arise when using AWS as the underlying cloud provider:
+
+* Kubernetes creates classic load balancers for LoadBalancer-type services,
+  which do not support IPv6.
+
+* Juju does not honor the "automatically assign IPv6 address" setting and
+  creates instances without IPv6 addresses. You can attach IPv6 addresses
+  after deploying with something like:
+
+  ```bash
+  for machine in $(juju status --format=json | jq -r '.machines|keys[]' | sort -n); do
+      echo -n "Machine $machine: "
+      instance="pending"
+      while [[ "$instance" == "pending" ]]; do
+          instance=$(juju status --format=json | jq -r '.machines["'"$machine"'"]."instance-id"')
+      done
+      interface=$(aws ec2 describe-instances --instance-id "$instance" --output text --query 'Reservations[*].Instances[*].NetworkInterfaces[*].NetworkInterfaceId')
+      aws ec2 modify-instance-attribute --instance-id "$instance" --no-source-dest-check
+      ipv6_addresses=$(aws ec2 describe-instances --instance-id "$instance" --output text --query 'Reservations[*].Instances[*].NetworkInterfaces[*].Ipv6Addresses')
+      if [[ -n "$ipv6_addresses" ]]; then
+          echo "$ipv6_addresses"
+      else
+          aws ec2 assign-ipv6-addresses --network-interface-id "$interface" --ipv6-address-count 1 --output text --query 'AssignedIpv6Addresses'
+      fi
+  done
+  ```
+
+### OpenStack
+
+Overall OpenStack is pretty straightforward.  I only encountered one issue
+which may be resolvable with some tweaks to the OpenStack environment:
+
+* Neutron LBaaS didn't seem to work well with an IPv6 LB, but this may have
+  been a configuration issue.
```

Comment:
Turns out we didn't expect Neutron LB to work with K8s anyway, so I'm working on testing it with Octavia and will update this section accordingly.